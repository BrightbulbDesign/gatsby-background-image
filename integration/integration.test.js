/**
 * @jest-environment node
 */

import '@testing-library/react/dont-cleanup-after-each'

const GBI = require('gatsby-background-image').default
const GBIES5 = require('gatsby-background-image-es5').default

describe(`<BackgroundImage /> in gatsby-background-image-es5`, () => {
  it(`should create new BackgroundImage instance`, () => {
    const component = new GBIES5({
      fadeIn: true,
    })
    expect(component).toMatchSnapshot()
  })
})

describe(`<BackgroundImage /> in gatsby-background-image`, () => {
  it(`should create new BackgroundImage instance`, () => {
    const component = new GBI({
      fadeIn: true,
    })
    expect(component).toMatchSnapshot()
  })
})
