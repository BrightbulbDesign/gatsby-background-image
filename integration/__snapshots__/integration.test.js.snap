// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackgroundImage /> in gatsby-background-image should create new BackgroundImage instance 1`] = `
BackgroundImage {
  "backgroundStyles": Object {
    "backgroundPosition": "center",
    "backgroundRepeat": "no-repeat",
    "backgroundSize": "cover",
  },
  "bgImage": "",
  "cleanUpListeners": undefined,
  "context": undefined,
  "handleImageLoaded": [Function],
  "handleRef": [Function],
  "imageRef": null,
  "intersectionListener": [Function],
  "props": Object {
    "fadeIn": true,
  },
  "refs": Object {},
  "selfRef": null,
  "state": Object {
    "IOSupported": false,
    "addedClassName": "",
    "currentClassNames": "",
    "fadeIn": true,
    "hasNoScript": true,
    "imageState": 0,
    "imgLoaded": false,
    "isVisible": false,
    "seenBefore": false,
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;

exports[`<BackgroundImage /> in gatsby-background-image-es5 should create new BackgroundImage instance 1`] = `
BackgroundImage {
  "backgroundStyles": Object {
    "backgroundPosition": "center",
    "backgroundRepeat": "no-repeat",
    "backgroundSize": "cover",
  },
  "bgImage": "",
  "cleanUpListeners": undefined,
  "context": undefined,
  "handleImageLoaded": [Function],
  "handleRef": [Function],
  "imageRef": null,
  "intersectionListener": [Function],
  "props": Object {
    "fadeIn": true,
  },
  "refs": Object {},
  "selfRef": null,
  "state": Object {
    "IOSupported": false,
    "addedClassName": "",
    "currentClassNames": "",
    "fadeIn": true,
    "hasNoScript": true,
    "imageState": 0,
    "imgLoaded": false,
    "isVisible": false,
    "seenBefore": false,
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
